app/src/test/java/com/ciandt/dragonfly/example/features/realtime/RealTimePresenterTest.kt

```diff
@@ -0,0 +1,120 @@
+package com.ciandt.dragonfly.example.features.realtime
+
+import com.ciandt.dragonfly.example.config.PreferenceKeys
+import com.ciandt.dragonfly.example.infrastructure.PreferencesRepository
+import com.nhaarman.mockito_kotlin.atLeastOnce
+import com.nhaarman.mockito_kotlin.eq
+import com.nhaarman.mockito_kotlin.never
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.whenever
+import org.amshove.kluent.any
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+
+
+class RealTimePresenterTest {
+
+    lateinit var presenter: RealTimePresenter
+
+    @Mock
+    lateinit var view: RealTimeContract.View
+
+    @Mock
+    lateinit var preferencesRepository: PreferencesRepository
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+
+        presenter = RealTimePresenter(preferencesRepository)
+    }
+
+    @After
+    fun tearDown() {
+        presenter.detachView()
+    }
+
+    @Test
+    fun attachView() {
+
+        whenever(
+                preferencesRepository.getBoolean(eq(PreferenceKeys.REAL_TIME_PERMISSIONS_PERMANENTLY_DENIED), any())
+        ).thenReturn(false)
+
+        presenter.attachView(view)
+
+        verify(view).checkRealTimeRequiredPermissions()
+    }
+
+    @Test
+    fun attachViewWithPermissionsDenied() {
+
+        whenever(
+                preferencesRepository.getBoolean(eq(PreferenceKeys.REAL_TIME_PERMISSIONS_PERMANENTLY_DENIED), any())
+        ).thenReturn(true)
+
+        presenter.attachView(view)
+
+        verify(view).showPermissionsRequiredAlert(any(), any(), any())
+    }
+
+    @Test
+    fun onRealTimePermissionsDenied() {
+
+        whenever(
+                preferencesRepository.getBoolean(eq(PreferenceKeys.REAL_TIME_PERMISSIONS_PERMANENTLY_DENIED), any())
+        ).thenReturn(false)
+
+        presenter.attachView(view)
+
+        verify(view).checkRealTimeRequiredPermissions()
+
+
+        presenter.onRealTimePermissionsDenied(false)
+
+        verify(preferencesRepository, never()).putBoolean(PreferenceKeys.REAL_TIME_PERMISSIONS_PERMANENTLY_DENIED, true)
```

Comment:
Line 76: Permissions were requested and the user denied, but not permanently. `onRealTimePermissionsDenied(permanently: Boolean)`

Now, at line 78, we check that `preferencesRepository.putBoolean(PreferenceKeys.REAL_TIME_PERMISSIONS_PERMANENTLY_DENIED, true)` is **never** called.